clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTaskTest.java

```diff
@@ -0,0 +1,197 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import static org.mockito.AdditionalMatchers.*;
+import static org.mockito.Mockito.*;
+
+import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper;
+import com.linkedin.davinci.config.VeniceStoreVersionConfig;
+import com.linkedin.davinci.replication.RmdWithValueSchemaId;
+import com.linkedin.davinci.replication.merge.RmdSerDe;
+import com.linkedin.davinci.stats.AggHostLevelIngestionStats;
+import com.linkedin.davinci.stats.HostLevelIngestionStats;
+import com.linkedin.davinci.storage.StorageEngineRepository;
+import com.linkedin.davinci.store.AbstractStorageEngine;
+import com.linkedin.venice.meta.ReadOnlySchemaRepository;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.schema.rmd.RmdSchemaEntry;
+import com.linkedin.venice.schema.rmd.RmdSchemaGenerator;
+import com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class ActiveActiveStoreIngestionTaskTest {
+  /**
+   * This test verifies that the IngestionReplicationMetadataLookUpLatency metric is always emitted regardless of
+   * traffic. This is useful in setting up health rules against this metric.
+   */
+  private static final String VALUE_SCHEMA_STR = "{" + "   \"type\" : \"record\","
+      + "   \"namespace\" : \"com.linkedin.avro\"," + "   \"name\" : \"Person\"," + "   \"fields\" : ["
+      + "      { \"name\" : \"Name\" , \"type\" : \"string\", \"default\" : \"unknown\" },"
+      + "      { \"name\" : \"Age\" , \"type\" : \"int\", \"default\" : -1 },"
+      + "      { \"name\" : \"Items\" , \"type\" : {\"type\" : \"array\", \"items\" : \"string\"}, \"default\" : [] },"
+      + "      { \"name\" : \"PetNameToAge\" , \"type\" : [\"null\" , {\"type\" : \"map\", \"values\" : \"int\"}], \"default\" : null }"
+      + "   ]" + "}";
+  private ActiveActiveStoreIngestionTask taskUnderTest;
+  private PartitionConsumptionState pcsMock;
+  private HostLevelIngestionStats hostLevelIngestionStats;
+  private AbstractStorageEngine storageEngineMock;
+  private ReadOnlySchemaRepository schemaRepository;
+  private final byte[] keyBytes = "key".getBytes();
+
+  private void setupMocksPre() {
```

Comment:
nit: we can use [depends method](https://testng.org/doc/documentation-main.html#dependent-methods) from testNG to make the tests cleaner in terms of execution ordering and we can also use more meaningful naming here.